docs/reference/mapping/runtime.asciidoc

```diff
@@ -0,0 +1,603 @@
+runtime field from a search request to the index mapping by moving the field
+definition from `runtime_mappings` in the search request to the `runtime`
+section of the index mapping.
+
+[[runtime-shadowing-fields]]
+=== Shadowing fields
+If you create a runtime field with the same name as a field that
+already exists in the mapping, the runtime field shadows the mapped field. At
+search time, {es} evaluates the runtime field, calculates a value based on the
+script, and returns the value as part of the query. Because the runtime field
+shadows the mapped field, you can modify the value returned in search without
+modifying the mapped field.
+
+If you define a runtime field that does not include a script, {es} evaluates the
+field at search time, looks at each document containing that field, retrieves
+the `_source`, and returns a value if one exists.
+
+As mentioned in <<runtime-mapping-fields,mapping a runtime field>>, you can
+shadow field values inside objects. For example, index the following documents into `my-index`:
+
+[source, console]
+----
+POST my-index/_bulk?refresh=true
+{"index":{}}
+{"name":{"first":"Jose","last":"Hickman"}}
+{"index":{}}
+{"name":{"first":"India","last":"Avila","suffix":"II"}}
+----
+
+In the `runtime_mappings` section of the `_search` request, you can define a
+script that operates on the `name.first` field. The values you specify will
+shadow field values in the index mapping for the duration of the query without
+modifying the field itself.
+
+The following request defines a runtime field that retrieves values based on
+the script valuation. The field defined in the search request shadows the
+indexed `name.first` field and substitutes a value for that field based on the
+logic defined in the script:
+
+[source,console]
+----
+POST my-index/_search
+{
+  "runtime_mappings": {
+    "name.first": {
+      "type": "keyword",
+      "script": {
+        "source": "if (\"Hickman\".equals(doc[\"name.last.keyword\"].value)) {emit(\"Guile\");} else if (\"Avila\".equals(doc[\"name.last.keyword\"].value)) {emit(\"Anastasia\");}"
+      }
+    }
+  },
+  "query": {
+    "match": {
+      "name.first": "Anastasia"
+    }
+  }
+}
+----
+// TEST[continued]
+
+
+[[runtime-retrieving-fields]]
+=== Retrieving a runtime field
+Use the <<search-fields,`fields`>> parameter on the `_search` API to retrieve
+the values of runtime fields. Runtime fields won't display in `_source`, but
+the `fields` API works for all fields, even those that were not sent as part of
+the original `_source`.
+
+The following request uses the search API to retrieve the `day_of_week` field
+that <<runtime-mapping-fields,this previous request>> defined as a runtime field
+in the mapping. The value for the `day_of_week` field is calculated dynamically
+at search time based on the evaluation of the defined script.
+
+[source,console]
+----
+GET my-index/_search
+{
+  "fields": [
+    "@timestamp",
+    "day_of_week"
+  ],
+  "_source": false
+}
```

Comment:
shall we also add the response for this request, to show how the runtime field is returned? Or is that redundant?